src/main/java/seedu/address/ui/CommandBox.java

```diff
@@ -82,4 +108,80 @@ public interface CommandExecutor {
         CommandResult execute(String commandText) throws CommandException, ParseException;
     }
 
+    /**
+     * Contains all successfully executed commands from the last application start.
+     * Keeps track of the currently displayed command's index in the history.
+     */
+    class CommandHistory {
+        private ArrayList<String> commandHistory;
+        private int index;
+
+        public CommandHistory() {
+            commandHistory = new ArrayList<>();
+            commandHistory.add("");
+
+            index = 0;
+        }
+
+        /**
+         * Adds a successfully executed command into {@code commandHistory}, and sets
+         * {@code textField} to an empty string.
+         *
+         * @param command the user-inputted String of the successfully executed command.
+         */
+        public void addCommand(String command, TextField textField) {
+            addToCommandHistory(command);
+            initialiseForNextCommand();
+            textField.setText("");
+
+            index = commandHistory.size() - 1;
+        }
+
+        private void addToCommandHistory(String command) {
+            commandHistory.set(commandHistory.size() - 1, command);
+        }
+
+        private void initialiseForNextCommand() {
+            commandHistory.add("");
+        }
+
+        private boolean isMostRecentCommand() {
+            return index == commandHistory.size() - 1;
+        }
+
+        private boolean noPreviousCommand() {
+            return index <= 0;
+        }
+
+        private boolean noNextCommand() {
+            return index >= commandHistory.size() - 1;
+        }
+
+        /**
+         * Sets the text in {@code textField} to the previous command, if a previous command exists.
+         * If the current command is the most recent command, adds that to history.
+         *
+         * @param textField the TextField where the user inputs commands in.
+         */
+        public void getPreviousCommand(TextField textField) {
+            if (noPreviousCommand()) {
+                return;
+            } else if (isMostRecentCommand()) {
+                addToCommandHistory(textField.getText());
+            }
```

Comment:
This will cause the current textfield to be inserted into the history even if its not entered right? I feel it will be more intuitive to not add this into the history list, but just have it set to a currentTextField variable, so we can get it back again if we do getNextCommand

Reply:
I think the main reason why I used this was because of the `index` field. If we don't represent the current command (that the user has not executed) in the command history, then your index would be equal to `commandHistory.size()`, which is out of bounds.

But I have pushed some commits to do this change! See which makes more sense